Any app setting CONFIG_PM or CONFIG_POWER_OFF will fail to build during cmake step on esp32c6_devkitc

**Describe the bug**
* tests/lib/cpp/cxx/cpp.main.cpp14 
* tests/lib/cpp/cxx/cpp.main.cpp17 
* tests/lib/cpp/cxx/cpp.main.cpp20 
* tests/lib/cpp/cxx/cpp.main.cpp2A 
* tests/lib/cpp/cxx/cpp.main.cpp2B 
* tests/lib/cpp/cxx/cpp.main.cpp98 
* tests/lib/cpp/cxx/cpp.main.minimal 
* tests/lib/cpp/cxx/cpp.main.newlib 
* tests/lib/cpp/cxx/cpp.main.newlib_nano 
* tests/lib/cpp/cxx/cpp.main.picolibc 

or 
Any test/sample which sets CONFIG_PM or CONFIG_POWER_OFF
will fail to build for esp32c6_devkitc

**To Reproduce**
Steps to reproduce the behavior:
1.  twister -p esp32c6_devkitc -T tests/lib/cpp/cxx/
4. See error

OR just
1. cmake -GNinja -DBOARD=esp32c6_devkitc ../samples/hello_world/ -DCONFIG_PM=y
2. See error

**Expected behavior**
No build failures. Tests pass

**Impact**
Failures in main CI

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/runs/27431582117
```
Cmake build failure
Loading Zephyr default modules (Zephyr base).
...
CMake Error at /__w/zephyr/zephyr/cmake/modules/extensions.cmake:428 (add_library):
  Cannot find source file:

    /__w/zephyr/zephyr/soc/espressif/esp32c6/power.c
```

**Environment (please complete the following information):**

 - OS: Linux, Ubuntu 22.04, CI
 - Toolchain: 0.16.8
 - Commit SHA or Version used: Main as of now fd0e46506fabd4c0c91ecb336a39cb6a6aa2b270 (3.7.0-rc3)

**Additional context**
Failures in CI bisected to 65f63622e273c19883b866d71a13e8f18b7d5c6c 
#70403
Underlying SOC issue present since its introduction in b3523c9bfaa6653b2f1bd60480a0febd5ceb8dd1

--- a/soc/espressif/esp32c6/Kconfig
+++ b/soc/espressif/esp32c6/Kconfig
@@ -14,8 +14,6 @@ config SOC_SERIES_ESP32C6
 	select RISCV_ISA_EXT_ZICSR
 	select HAS_ESPRESSIF_HAL
 	select XIP if !MCUBOOT
-	select HAS_PM
-	select HAS_POWEROFF
 
 if SOC_SERIES_ESP32C6
 

